### PR TITLE
apps/examples/testcase/ta_tc/audio/itc: Add ITCs for Audio_Framework Module

### PR DESCRIPTION
**Pull request by arvinmittal**
@ apps/examples/testcase/ta_tc/audio/itc , Add ITCs for audio_framework

Signed-off-by: Arvin Mittal <arvin.mittal@samsung.com>

### CODE REVIEW

**Comment by arvinmittal** [comment]
Added ITCs for Audio_framework module, Please review
@jeongchanKim 
@sunghan-chang 
@junmin-kim

**Comment by lokeshbv** [comment]
Please use tab indentation.
FYI https://wiki.tizen.org/Native_Platform_Coding_Idiom_and_Style_Guide#Indentation

**Comment by arvinmittal** [comment]
Fixed review comments for audio_framework ITCs
@jeongchanKim 
@sunghan-chang 
@junmin-kim

**Comment by sunghan-chang** [comment]
@arvinmittal
1. Here is only Tizen RT project. Don't add [TizenRT] prefix on commit message. 
2. To know what you modify easily and quickly on commit message, let's add prefix for module like belows.
```
apps/testcase/audio: Added ITCs
or
apps/testcase: Added ITCs for Audio_Framework Module
```

File: apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c [hunk 1]
```diff
@@ -0,0 +1,543 @@
...
+	char *str = NULL;
+	g_pcm = pcm_open(0, 0, PCM_IN, NULL);
+
+	TC_ASSERT_GT("pcm_open", pcm_get_file_descriptor(g_pcm), 0);
+
+	size = pcm_get_buffer_size(g_pcm);
+	rate = pcm_get_rate(g_pcm);
+	bytes = rate * (pcm_format_to_bits(pcm_get_format(g_pcm)) / 8) * pcm_get_channels(g_pcm) * AUDIO_RECORD_DURATION;
+
+	g_record_buffer = (char *)malloc(bytes);	//rate * bit * channels * duration
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c]: please check utc. utc doesn't use global buffer anymore...

**Comment by arvinmittal** [comment]
@ apps/examples/testcase/ta_tc/audio/itc , Fix review comments for audio_framework ITCs
Please review.
@jeongchanKim
@sunghan-chang
@junmin-kim

**Comment by sunghan-chang** [comment]
Don't you read comment? I write it again.
@arvinmittal
1. Here is only Tizen RT project. Don't add [TizenRT] prefix on commit message. 
2. To know what you modify easily and quickly on commit message, let's add prefix for module like belows.
```
apps/testcase/audio: Added ITCs
or
apps/testcase: Added ITCs for Audio_Framework Module
```

File: apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c [hunk 2]
```diff
@@ -0,0 +1,551 @@
...
+ ****************************************************************************/
+
+/// @file tc_audio_main.c
+/// @brief Main Function for Audio TestCase Example
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdio.h>
```
Comment (sunghan-chang) [inline apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c]: To use CONFIG_XX, you should include <tinyara/config.h>

File: apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c [hunk 3]
```diff
@@ -0,0 +1,551 @@
...
+#include <stdlib.h>
+#include <unistd.h>
+#include <pthread.h>
```
Comment (sunghan-chang) [inline apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c]: why do you include pthread.h?

File: apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c [hunk 4]
```diff
@@ -0,0 +1,551 @@
...
+#include <dirent.h>
+#include <fcntl.h>
+#include <string.h>
+#include <errno.h>
```
Comment (sunghan-chang) [inline apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c]: same as above

**Comment by arvinmittal** [comment]
apps/examples/testcase/ta_tc/audio/itc: Fix review comments for audio_framework ITCs
Please review.
@jeongchanKim
@sunghan-chang
@junmin-kim

File: apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c (hunk 4 above)
Comment (arvinmittal) [reply apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c]: @sunghan-chang 
Removed unused header files from the itc_audio_main.c testcase file. Also added  <tinyara/config.h> header file.
Not removed fcntl.h header as it is required in testcase.

File: apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c [hunk 5]
```diff
@@ -0,0 +1,548 @@
...
+
+	/* use default config here */
+	g_pcm = pcm_open(0, 0, PCM_OUT, NULL);
+	TC_ASSERT_GT_CLEANUP("pcm_writei", pcm_is_ready(g_pcm), 0, close(fd));
+
+	size = pcm_get_buffer_size(g_pcm);
+	rate = pcm_get_rate(g_pcm);
+	bytes = rate * (pcm_format_to_bits(pcm_get_format(g_pcm)) / 8) * pcm_get_channels(g_pcm) * AUDIO_RECORD_DURATION;
+
+	buffer = (char *)malloc(bytes);	//rate * bit * channels * duration
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c]: please check utc this will takes huge size buffer.

Comment (arvinmittal) [reply apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c]: @Taejun-Kwon  Updated testcase as per request. Please check

File: apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c [hunk 6]
```diff
@@ -0,0 +1,543 @@
...
+	g_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_GT("pcm_open", pcm_get_file_descriptor(g_pcm), 0);
+
+	pcm_set_config(g_pcm, NULL);
+	size = pcm_get_buffer_size(g_pcm);
+	TC_ASSERT_GT("pcm_get_buffer_size", size, 0);
+
+	frame_size = pcm_bytes_to_frames(g_pcm, size);
+	TC_ASSERT_GT("pcm_bytes_to_frames", frame_size, 0);
+	TC_SUCCESS_RESULT();
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c]: please add TC_ASSERT_EQ("pcm_close", pcm_close(g_pcm), 0);

File: apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c [hunk 7]
```diff
@@ -0,0 +1,543 @@
...
+	pcm_set_config(g_pcm, NULL);
+	size = pcm_get_buffer_size(g_pcm);
+	TC_ASSERT_GT("pcm_get_buffer_size", size, 0);
+
+	frame_size = pcm_bytes_to_frames(g_pcm, size);
+	TC_ASSERT_GT("pcm_bytes_to_frames", frame_size, 0);
+
+	bytes = pcm_frames_to_bytes(g_pcm, frame_size);
+	TC_ASSERT_GT("pcm_frames_to_bytes", bytes, 0);
+	TC_SUCCESS_RESULT();
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c]: please add TC_ASSERT_EQ("pcm_close", pcm_close(g_pcm), 0);

File: apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c [hunk 8]
```diff
@@ -0,0 +1,543 @@
...
+{
+	int fd = 0;
+	char input_str[INPUT_STR_LEN];
+	int ret = 0;
+	char *str = NULL;
+	char *buffer = NULL;
+	unsigned int bytes_per_frame =0;
+	int frames_read = 0;
+	int remain = 0;
+
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c]: please add	g_pcm = pcm_open(0, 0, PCM_IN, NULL);

Comment (arvinmittal) [reply apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c]: @Taejun-Kwon 
Fixed your review comments, Please check

**Pull request merged**